### ISSUE TITLE
Sending your pet in attack dont make you immune to sap

### ISSUE THREAD

**Issue by Winns** [ccshiro/cc-buglist#246]
Sending you pen in attack should make you immune to sap. [gif from corecraft](https://github.com/ccshiro/cc-buglist/issues/215)

**How its work**
Command your pet to attack target, while pet is still running to target but do not reach him yet you should be immune to sap, also you wont get "in combat" marker until pet start attack target.

This was one of the core warlock/hunter move against rogue teams, so dont rate this as a bug that should not exist, game was balanced around this.

Based on my own experience as an active warlock arena player of s7,s8,s9.
Also found some threads about this -
http://www.arenajunkies.com/topic/118924-combat-bug-with-pet/
http://www.arenajunkies.com/topic/154952-pet-combat-sap-bug/
http://www.arenajunkies.com/topic/72876-pet-combat-bug/
http://www.arenajunkies.com/topic/103477-3v3-warlock-sap-bug/#entry1375040
[Cataclysm: Pet attack makes the game see you as in combat, meaning you are unsappable even if your pet never struck the target.](http://www.arenajunkies.com/topic/180753-warlock-cataclysm-warlock-guide-for-the-contest/)

**Comment by braide** [comment]
No, you're not immune to sap when you send your pet to attack. In TBC your immune when you gain combat, and you only gain combat when the pet actually lands an attack or spell. That is how it was s1,s2,s3,s4. The threads you have shown are from 2009 (which is WOTLK). And your experience is from s7 and foward, which again is not TBC.

**Comment by Winns** [comment]
I did not find any confirmation that this mechanic was changed in WOTLK, found only this comment ["..it was discovered in TBC"](http://www.arenajunkies.com/topic/103477-3v3-warlock-sap-bug/#entry1375040). Gonna check warlock arena videos later.

**Comment by MorbidMike** [comment]
braide is right. I remember actually sending my pet out to Devour Magic on Mages to prevent getting sapped. Simply sending it to attack would not do the trick.

**Comment by solowt** [comment]
Braide is correct.  In TBC you could get into combat with your pet but it involved your pet actually taking damage, dealing damage, or casting a spell on an enemy player.

**Comment by KuroiKensai** [comment]
Author is wrong. You are immune to sap only when you are in combat and when your pet is just running - you are not in combat and not immune. If author dont have videos from official 2.4.3 server proofing this of course. Even at the link he provided man clearly states: "it puts them IN COMBAT and they can't be sapped". Even if such thing really existed - this is obviously bug and abuse.

**Comment by Klimpb** [comment]
I know that this existed pretty much through the entire wrath of the lich king because it was the most important thing when playing beastcleave for an enhancement shaman, get wolfs out and send them to attack and you would never get sapped since it put you in combat instantly. I assume it was the same with hunter/warlock pets.

I'm not sure if it was a bug or not though since it was quite a well known thing.

That being said, I remember no such thing from TBC. But I might be wrong.

**Comment by solowt** [comment]
People sent in pets to get in combat in TBC.  But the pets actually had to reach the enemy or cast a spell on them.

**Comment by Devilblessed** [comment]
When you send your pet to attack you're not immune to sap. You are immune to sap when you ENTER combat just like all the other classes, and you enter combat when your pet reaches the enemy and attacks.

**Comment by Klimpb** [comment]
Proof: http://webmup.com/Bt5RD/ (look at his water on the actionbar)

This report is invalid.

**Comment by Stoney21** [comment]
It's full of shit report. You get combat if you make an action with your pet on somebody AND ONLY THEN. No action = no combat. You can't get in combat by simply sending your pet to attack someone, you played too much on Smoldershit.

**Comment by Winns** [comment]
SuddenDeath video start when warlock already sapped, do you have a full video ? "look at his water on the actionbar" - what do you mean by that ?

Stoney21 dont need to post here If you're incompetent in pvp mechanics.

This report here is to get a truth about _tbc_ combat mechanic, I have provided my facts that this move existed through whole _wotlk_, i want to see some facts or at least personal tbc experience against or for this statement relatively to _tbc_.

**Comment by Klimpb** [comment]
Winns by water I mean this: 
![](http://i.gyazo.com/921d068bbd3330a5beb7910356f6c39d.png)

When it becomes unusable it is a sign of entering combat. Which did not happen until the felhunter actually casted spell lock on the mage.

**Comment by Winns** [comment]
Ok i see, but combat state is displayed at player portrait (level) you dont need water for this.
This "unsappable" move work a bit different.

> **How it work**
> Command your pet to attack target, while pet is still running to target but do not reach him yet you should be immune to sap, also you wont get "in combat" marker until pet start attack target.

You wont get "in combat" marker until pet start attack, but you are already immune to sap when he is just run to target.

**Comment by solowt** [comment]
Are you high?  Being in combat is what makes you immune to sap.  There's no separate mechanic to make you immune to sap but not in combat.  

I don't know (or care) if this changed in wotlk+.  This is how it worked in vanilla and TBC.

**Comment by somnu** [comment]
I actually remember what OP says and it was clearly happening in TBC, (warlocks somehow got into combat by just putting the pet to attack my ally who was 100% out of los of pet or lock) but I never bothered to understand how it worked back then. I just raged when they did it cause it was clearly an abuse of an already overpowered class and I'm happy it won't be implemented on corecraft. In wotlk it was fixed and didn't happen anymore, and the only way to get in combat was to actually make the pet cast devour magic or attack on a target. Also it wasn't a separate mechanic that made you immune to sap, it just said "target is in combat".
Source: I played rogue all throughout TBC and WOTLK.

**Comment by vexxedd** [comment]
> I actually remember what OP says and it was clearly happening in TBC, (warlocks somehow got into combat by just putting the pet to attack my ally who was 100% out of los of pet or lock) but I never bothered to understand how it worked back then.

Yes, but it was a bug. Back then, (guess it was 2008) there was some tournament (blizzcon?). Do you remember Inflame/Neilyo/Serennia lock/rogue/druid team? They also switch to some war/war/pala in random games. There was also funny video about Neilyo teaching some guy how to wrestling. Back to the point. Inflame was abusing that shit in whole tournament. Basically what you do was something like this: command pet to attack / rain of fire / healthfunell ur pet and you got combat. But was fixed shortly after, so not the point. 
You cant get immune to sap/get combat unless your pet attack/spell lock/devour enemy player. Period.

**Comment by Stoney21** [comment]
You are fucking retarded and you should kill yourself if you think you should get in combat by simply sending a pet on someone and befor it even HITS anything. That way you could target any rogue/druid as soon as arena starts and send your pet on them so they can't stealth. IT WOULDN'T MAKE ANY SENSE SO GET A BRAIN BEFOR REPORTING ANYTHING.

**Comment by ghost** [comment]
The sensible behaviour is for pet hostile actions to put owner in combat, 'pet attack' shouldn't count regardless of the state of TBC and when it was fixed by blizzard.

Someone might argue that this will tip the pvp balance away from pet teams. Well it's not game-changing bug that someone 100% needs to abuse in order to win. If they relied on abusing such bug then why would we care?

Yes, bugs existed back in TBC, does that mean CC should implement them all? Think not

**Comment by somnu** [comment]
I think vexxedd nailed it, and yes - it was a bug of course it was. "Fixing" stuff like this would be like purposely reintroducing exploits into the game, which isn't blizzlike, it's just dumb.

**Comment by jezbelle** [comment]
Hold up. Just so you guys know- the icon of the water in the picture is for WATER OF THE STARS. Works a little differently than regular water in arena possibly. Maybe not. I just wanted to throw that out there.

**Comment by solowt** [comment]
Star's Tears work the same as normal water except normal water (except mage water) can't be used in  arenas.

**Comment by meowmewo** [comment]
http://www.arenajunkies.com/topic/103477-3v3-warlock-sap-bug/#entry1375040
http://www.arenajunkies.com/topic/79011-death-knight-evading-sap/
http://www.skill-capped.com/forums/archive/index.php/t-2749.html

It seems it really was like that back in TBC and early Wrath .Read everything, From what i understood is that if ur pet is on the way u cant be sapped BUT ur not in combat as well.If a healer is healing u at that time he wont get in combat and he can be sapped but not you.If u call ur pet back u can be sapped again.

**Comment by solowt** [comment]
All of those posts are from early wrath.

**Comment by meowmewo** [comment]
i told u Read everything all comments.READ

**Comment by Brumeux** [comment]
Why are you guys bashing on him ? It's not a bug that use to happen on retail server. I have no idea about tbc but it has been like that whole wotlk. Do you think it would have stayed this way that long if it was a bug ? No, it was clearly intentional. It wasn't an abuse nor anything, anyway most locks didnt know that. 
And just to clarify: NO it wouldn't put you in combat until the pet actually reaches the target or casts a spell. So @Stoney21 stop the caps abuse and YOU get a brain. Rogues/droods would be able to stealth, they wouldn't be in combat. Only thing is that the lock would be IMMUNE to sap.

**Comment by exltus** [comment]
[here](https://www.youtube.com/watch?v=5Ixfivc3AoU#t=54) is video, that is showing warlock sending his pet and you can clearly see, that he is not getting combat for almost 5 secs. Pet even casted Devour Magic, that was reflected and even it did not put warlock in combat. So this issue is fully INVALID

**Comment by lawzftw** [comment]
Because you never understood PvP mechanics doesn't mean those mechanics don't exist.

I know for sure how it worked during WoTLK (on retail), and I'm going to explain it :
- If you send your pet to an enemy, you can't be sapped, regarding your combat state. It is NOT related to your combat state, which means you don't have those "crosse blades" and you still can not be sapped.
- If your pet hits an enemy with a melee hit, or a damaging ability, he will enter combat, and the owner of the pet as well.
- If your pet hits an enemy with a non damaging ability (devour magic, soothing kiss etc), he will enter combat, but the owner will not (and can be sapped if pet isn't running to the enemy)
- If your pet gets hit, he will gain combat, but the owner won't (and can be sapped if pet isn't running to the enemy)
- If the owner of the pet uses health funnel and pet is in combat, owner will gain combat.

I tested it when on retail wotlk, which is the problem (wotlk). But I saw too much nonsense in this thread, and it can still be useful in a way (at least ppl who tell bullshit like "looool noobs if lock isn't in combat he can be sapped!1! even during wotlk").

The only "proof" you can find to say this mechanism didn't exist during TBC is finding a video with a warlock sending his pet, and still getting sapped. It shouldn't be that hard, considering they often send pet to gain combat, right ? Btw don't forget rogues could "bug" pet (even on retail) with distract, so if the pet stops moving mid map, and lock gets sapped, it doesn't count.

**Comment by Philosophers** [comment]
I saw this topic and (as a hunter) I need to comment.

First of all:
@lawzftw "The only "proof" you can find to say this mechanism didn't exist during TBC is finding a video with a warlock sending his pet, and still getting sapped."

You have to proof you're right, not others have to proof you're wrong. Not true until proven true. Like not guilty until proven guilty.

I played as a hunter on retail during TBC: I do recall that I was using Wind Serpent as a pet in the arenas just because of it's 20yd range ligtning breath catching combat, to not only rely on trap+flare, cause it was imperfect for skilled rogues. + It can end enemy drinking earlier. I also recall that I dropped Wind Serpent since release of wotlk (I am not sure why, but changes: that could be it). I am pretty sure that hunter pet works same as warlock pet.

I am not sure what it was like that on wotlk, but this expansion did big changes to almost every in-game mechanincs, so anything with "on wotlk" is not proving anything.

**Comment by lawzftw** [comment]
@Philosophers This is not the way it should be done for such a project. The aim is not to prove someone guilty, it is to get the best scripts for this server, balance wise as well. And this contributes to balance (it contributed to balance A LOT during wotlk at least). So if you truly believe in this server's aim, you shouldn't get that reasoning. This is not a trial. Not at all.

However, I've tried to find some evidence about that "mechanism" and I found a lot of tlk posts, even quite early tlk, but no tbc posts, so it sounds like it didn't work that way in 2.4.3. But a video proof would be the best (couldn't find any video about that :/)

**Comment by Philosophers** [comment]
@lawzftw That was metaphor. You have to prove that something is right before it can be changed. Before someone has actual proof there is nothing to discuss.

I am still sure that in TBC pets with ranged attacks were so useful because of that combat catching.

**Comment by exltus** [comment]
Hmmm... I think, that I have answer to this topic. I crawled whole web and didn´t find nothing about this issue before WOTLK release. If you put time time restrictions in your google search and you will search for "unsapable" you will get 0 results from 2004 to november 2008, but if you will do same thing in date range from 2009 till now you will get a ton of results. Here are some examples (note, that are from 3.1.0 and higher patch) [1](http://webcache.googleusercontent.com/search?q=cache:jOje713JL8EJ:www.arenajunkies.com/topic/143935-saping-locks/+&cd=1&hl=cs&ct=clnk&gl=cz) [2](https://github.com/TrinityCore/TrinityCore/issues/7567) [3](http://www.arenajunkies.com/topic/143935-saping-locks/) etc.
So it looks, that this change comes with WOTLK, that means, that is INVALID

**Comment by ccshiro** [comment]
@exltus that is a good way of going about it. I will leave this as needs research for a bit longer, in case someone feels like they want to make a counter-point, otherwise I will consider that determining and invalidate this issue.

**Comment by Trapstarx** [comment]
Playing rogue for ages and I can surely say that sending your pet in attack shouldn't put you in combat  until pet lands an attack or spell on target in TBC.

**Comment by Roarek** [comment]
Sending a pet in should not put you in combat until a combat action has been initiated (attacking, being attacked, casting a spell on the target, etc). In WOTLK this worked differently (where sending your pet in put you in combat) and that's what you're thinking of. It has never worked that way before WOTLK and neither has it worked that way again after WOTLK. There isn't a "sap immunity" functioning this way in BC. This is not a bug.

**Comment by Brumeux** [comment]
@Trapstarx @Roarek 
Either you've not read what we said, either you still don't get what's the point here. None is talking about being in COMBAT. Besides you are wrong. It wouldn't do so too in WOTLK.

Anyway I think @exltus proved how it worked back in tbc.

**Comment by Philosophers** [comment]
@exltus I agree, here is my part
I've been doing my research for hours. I have no proof of what I am going to say, but after I reviewed a lot of guides / posts I got these observations:

I read a lot of "issue" reports, wrote by rogues, who say that hunter / warlock / death knight is immune to sap, even if their pet did not reach its target. All of that reports were dated the time WoTLK was live (for a some time). Did not found any of such post preWoTLK.

In almost every single larger hunter / warlock guide from WoTLK (was not looking into DK guides) there is info about trick, that if you send pet to attack (it is mentioned that it's just enough to send it, it does not need to land a hit) you will be immune to sap. Macros, tips, etc. Meanwhile I did not found any of this "tips and tricks" from TBC guides.

I know that is a little sh*t argument, but it is hard to find straight evidence, cause it's about "hidden" mechanic of WoW and it's not listed in any official patch notes. I can provide links to 5-10 guides, both from TBC and WoTLK if you wanna check this theory.

PS: and yes, yes, yes, I know that guides are made by players and they can be wrong, etc. But if nobody ever reported about this before WoTLK it didn't exist, did it?

TL,DR:
Did not find anything about this working in TBC
Found a lot about this working in WoTLK.
